pom.xml

```diff
@@ -57,10 +57,39 @@
       <artifactId>application-licensing-licensor-api</artifactId>
       <version>${licensing.version}</version>
     </dependency>
+    <!-- Because adal4j is an archived repo, as part of https://github.com/xwikisas/application-office365/issues/26
+      we need to move away from it. Until then, some exclusions are added in order to override old dependencies versions
+      that are brought by it or by transitive dependencies. A comment is added for each dependency that should be
+      removed when this migration is done. -->
     <dependency>
       <groupId>com.microsoft.azure</groupId>
       <artifactId>adal4j</artifactId>
       <version>1.6.7</version>
+      <exclusions>
+        <!-- Exclude json-smart because the declared range is lower than the managed one in some cases. See
+          https://github.com/xwikisas/application-office365/issues/25. -->
+        <exclusion>
+          <groupId>net.minidev</groupId>
+          <artifactId>json-smart</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <!-- This should be removed once #26 is fixed. -->
+    <dependency>
+      <groupId>net.minidev</groupId>
+      <artifactId>json-smart</artifactId>
+      <version>2.4.11</version>
```

Comment:
How did you choose this version? Not that when you specify a single version, rather than a range, it means it needs that version or any newer version, so similar to ``[2.4.11,)``. What bothers us in #25 is the upper limit of the range. I mean, we don't have to specify the latest version, because this forces other extensions to use this version, unless we know the latest version fixes some issues that are affecting us. Otherwise we could simply use the lower limit of the range.

Now, the fact that they used a range usually signals that they knew it doesn't work with a more recent version, but you must have tested and you didn't notice any issue, right?

Reply:
Actually, to confirm this change, I noticed that the same exclusion was done on OIDC as well, since it uses the same `oauth2-oidc-sdk` library that ADAL4J brings transitively https://github.com/xwiki-contrib/oidc/blob/8c1012d9d0dfff6564ee0a5ba1749c43b4d2de8f/pom.xml#L52C29-L52C29